libkbfs/folder_branch_ops.go

```diff
@@ -539,20 +578,43 @@ func (fbo *folderBranchOps) setHeadLocked(
 		// revision is enough to trigger conflict resolution.
 		fbo.cr.Resolve(md.Revision(), MetadataRevisionUninitialized)
 	} else if md.MergedStatus() == Merged {
+		mergedRev := md.Revision()
+
 		journalEnabled := TLFJournalEnabled(fbo.config, fbo.id())
 		var key kbfscrypto.VerifyingKey
 		if journalEnabled {
-			key, err = fbo.config.KBPKI().GetCurrentVerifyingKey(ctx)
-			if err != nil {
-				return err
+			if isFirstHead {
```

Comment:
Now that the logic here is more complicated, it might be clearer to have each branch call `setLatestMergedRevisionLocked` appropriately, like:

```
if journalEnabled {
  if isFirstHead {
    ...
    fbo.setLatestMergedRevisionLocked(ctx, lState, journalPred, false)
  } else {
    key, err := ...
    if key.KID() != md.LastModifyingWriterKID() {
      fbo.setLatestMergedRevisionLocked(ctx, lState, md.Revision(), false)
    }
  }
} else {
  fbo.setLatestMergedRevisionLocked(ctx, lState, md.Revision(), false)
}
```